go-controller/pkg/ovn/policy.go

```diff
@@ -597,12 +445,14 @@ func (oc *Controller) createMulticastAllowPolicy(ns string, nsInfo *namespaceInf
 	}
 
 	// Add all ports from this namespace to the multicast allow group.
-	for _, portName := range nsInfo.addressSet {
-		if portInfo, err := oc.logicalPortCache.get(portName); err != nil {
-			klog.Errorf(err.Error())
-		} else if err := podAddAllowMulticastPolicy(ns, portInfo); err != nil {
-			klog.Warningf("failed to add port %s to port group ACL: %v", portName, err)
-		}
+	if as := oc.getAddressSetFromCache(ns); as != nil {
+		as.ForEachPort(func(portName string, _ net.IP) {
```

Comment:
let's not combine concepts of address sets and port groups. If you want to cache port group info too, that's good, let's use a port group type to hold it

Reply:
@trozet actually here isn't not port group, but logical switch port. But if I change it to a set of IPs, then this will change to ForEachIP so I guess that satisfies the request.